server/src/main/java/org/elasticsearch/reservedstate/service/ReservedStateVersion.java

```diff
@@ -47,17 +47,13 @@ public static ReservedStateVersion parse(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
 
-    public Version minCompatibleVersion() {
-        return compatibleWith;
-    }
-
     public static ReservedStateVersion readFrom(StreamInput input) throws IOException {
-        return new ReservedStateVersion(input.readLong(), Version.readVersion(input));
+        return new ReservedStateVersion(input.readLong(), BuildVersion.fromVersionId(input.readVInt()));
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeLong(version());
-        Version.writeVersion(compatibleWith(), out);
+        out.writeVInt(buildVersion().id());
```

Comment:
This will always use 5 bytes with the default serverless version - is it worth adding a special-case for small negative numbers here?

Reply:
Seems like that could be a followup, but it would require a new transport version?

Reply:
It would, yes